platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java

```diff
@@ -0,0 +1,399 @@
+        ((InputStreamContent) httpContent).setRetrySupported(true);
+        HttpRequest httpRequest = httpTransport.createRequestFactory()
+                .buildPostRequest(new GenericUrl(responseConsumerURL), httpContent);
+        HttpUnsuccessfulResponseHandler httpUnsuccessfulResponseHandler = (request, response, supportsRetry) -> {
+            String redirectLocation = response.getHeaders()
+                    .getLocation();
+            if (request.getFollowRedirects() && HttpStatusCodes.isRedirect(response.getStatusCode())
+                    && redirectLocation != null) {
+                // resolve the redirect location relative to the current location
+                request.setUrl(new GenericUrl(request.getUrl()
+                        .toURL(redirectLocation)));
+                // on 303 change method to GET
+                if (response.getStatusCode() == HttpStatusCodes.STATUS_CODE_SEE_OTHER) {
+                    request.setRequestMethod(HttpMethods.GET);
+                    // GET requests do not support non-zero content length
+                    request.setContent(null);
+                }
+                // remove Authorization and If-* headers
+                request.getHeaders()
+                        .setAuthorization((String) null);
+                request.getHeaders()
+                        .setIfMatch(null);
+                request.getHeaders()
+                        .setIfNoneMatch(null);
+                request.getHeaders()
+                        .setIfModifiedSince(null);
+                request.getHeaders()
+                        .setIfUnmodifiedSince(null);
+                request.getHeaders()
+                        .setIfRange(null);
+                request.getHeaders()
+                        .setCookie((String) ((List) response.getHeaders()
+                                .get("set-cookie")).get(0));
+                return true;
+            }
+            return false;
+        };
+        httpRequest.setUnsuccessfulResponseHandler(httpUnsuccessfulResponseHandler);
+        httpRequest.getHeaders()
+                .put("SOAPAction", "http://www.oasis-open.org/committees/security");
+        //has 20 second timeout by default
+        HttpResponse httpResponse = httpRequest.execute();
+        HttpResponseWrapper httpResponseWrapper = new HttpResponseWrapper();
+        httpResponseWrapper.statusCode = httpResponse.getStatusCode();
+        httpResponseWrapper.content = httpResponse.getContent();
+        return httpResponseWrapper;
+    }
+
+    @Override
+    public void handleFault(Message message) {
+
+    }
+
+    private String buildSoapMessage(String token, String relayState, Element body,
+            Response paosResponse) throws WSSecurityException {
+        String updatedMessage = soapMessage.replace("{{XmlBody}}", DOM2Writer.nodeToString(body));
+        if (token != null) {
+            updatedMessage = updatedMessage.replace("{{WSSecurity}}", securityHeader);
```

Comment:
This reads unclearly. At first I didn't realize what it was doing and thought you were trying to replace `{{token}}` in the `soapMessage` template which wasn't right. On second review, I see that you're inserting the `securityHeader` into `soapMessage` and then replacing its token. I think it would be clearer something like:
```java
String updatedSecHdr = securityHeader.replace("{{token}}", token);
updatedMessage = updatedMessage.replace("{{WSSecurity}}", updatedSecHdr);
```